manta-pay/src/crypto/hash/poseidon/parameter_generation/lfsr.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! LFSH implementation.
+// adapted from: https://github.com/arkworks-rs/sponge/blob/51d6fc9aac1fa69f44a04839202b5de828584ed8/src/poseidon/grain_lfsr.rs
+
+use alloc::vec::Vec;
+
+use crate::crypto::hash::ParamField;
+
+pub struct GrainLFSR {
+    state: [bool; 80],
+    prime_num_bits: u64,
+    head: usize,
+}
+
+fn append_bits<T: Into<u128>>(state: &mut [bool; 80], head: &mut usize, n: usize, from: T) {
+    let val = from.into() as u128;
+    for i in (0..n).rev() {
+        state[*head] = (val >> i) & 1 != 0;
+        *head += 1;
+        *head %= 80;
+    }
+}
+
+impl GrainLFSR {
+    pub fn new(prime_num_bits: u64, width: usize, r_f: usize, r_p: usize) -> Self {

```

Comment:
Add docs and use full names for these inputs.

Reply:
```suggestion
/// Return a new `GrainLFSR` for poseidon parameter generation.
pub fn new(prime_num_bits: u64, width: usize, full_rounds: usize, partial_rounds: usize) -> Self {   
```